processing/src/main/java/io/druid/query/UnionAllQuery.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import io.druid.query.spec.QuerySegmentSpec;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
+public class UnionAllQuery extends BaseQuery<Result<Object>>
+{
+  private final List<Query> queries;
+  private final boolean sortOnUnion;
+
+  @JsonCreator
+  public UnionAllQuery(
+      @JsonProperty("queries") List<Query> queries,
+      @JsonProperty("sortOnUnion") boolean sortOnUnion,
+      @JsonProperty("context") Map<String, Object> context
+  )
+  {
+    super(queries.get(0).getDataSource(), queries.get(0).getQuerySegmentSpec(), false, context);
```

Comment:
this will throw an NPE at early stage if `queries` is null. can we add a pre-check